skopt/sample.py

```diff
@@ -0,0 +1,79 @@
+from collections import Mapping
+from numbers import Integral
+from numbers import Real
+
+from scipy.stats._distn_infrastructure import rv_frozen
+from scipy.stats.distributions import randint
+from scipy.stats.distributions import uniform
+
+from sklearn.utils import check_random_state
+from sklearn.utils.fixes import sp_version
+
+
+class Categorical:
+    def __init__(self, *args):
+        self.values = args
+
+    def rvs(self, random_state=None):
+        rng = check_random_state(random_state)
+        return self.values[rng.randint(len(self.values))]
+
+    @property
+    def low(self):
+        return min(self.values)
+
+    @property
+    def high(self):
+        return max(self.values)
+
+
+class Discrete:
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+
+    def rvs(self, random_state=None):
+        return randint(self.low, self.high).rvs(random_state=random_state)
+
+
+class Continuous:
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+
+    def rvs(self, random_state=None):
+        return uniform(self.low, self.high).rvs(random_state=random_state)
+
+
+def points(grid, random_state=None):
```

Comment:
How about `random_points` as name? I didn't go for `sample_points` as it would be `sample.sample_points` which seems a bit silly.

Do we need to make this a class? I started with a class and didn't like having to define several double-underscore functions ... I like it as a generator.